### PR TITLE
[plugins/aws][feat] Collect Custom Domain Names (ApiGateway)

### PR DESCRIPTION
**Pull request by anjafr**
# Description

new resource:
- `aws_api_gateway_domain_name`

# To-Dos

- [x] Add test coverage for new or updated functionality
- [x] Lint and test with `tox`
- [ ] Document new or updated functionality (someengineering/resoto.com#XXXX)

# Code of Conduct

By submitting this pull request, I agree to follow the [code of conduct](https://resoto.com/code-of-conduct).

### CODE REVIEW

**Review by aquamatthias — COMMENTED** [review]
One note regarding no optional fields - otherwise lgtm. thanks!

File: plugins/aws/resoto_plugin_aws/resource/apigateway.py [hunk 1]
```diff
@@ -373,7 +376,7 @@ class AwsApiGatewayRestApi(ApiGatewayTaggable, AwsResource):
     api_binary_media_types: List[str] = field(factory=list)
     api_minimum_compression_size: Optional[int] = field(default=None)
     api_key_source: Optional[str] = field(default=None)
-    api_endpoint_configuration: Optional[AwsApiGatewayEndpointConfiguration] = field(default=None)
+    api_endpoint_configuration: AwsApiGatewayEndpointConfiguration = field(default=None)
```
[the remark this answers is not in the document]
Comment (anjafr) [reply plugins/aws/resoto_plugin_aws/resource/apigateway.py:379]: @aquamatthias Excellent question. I don't know. I wanted to remove that but I'm not allowed to have an attribute without default here so, uhh, I put it back 😬 🤷🏻‍♀️ 
This section is always present in the response, it's just the stuff inside that section that can vary. So it isn't optional but didn't find out how to define that properly

[the remark this answers is not in the document]
Comment (aquamatthias) [reply plugins/aws/resoto_plugin_aws/resource/apigateway.py:379]: I would opt to leave it optional - we rely on AWS to provide us a value.

[the remark this answers is not in the document]
Comment (anjafr) [reply plugins/aws/resoto_plugin_aws/resource/apigateway.py:379]: Ok sure

[the remark this answers is not in the document]
Comment (aquamatthias) [reply plugins/aws/resoto_plugin_aws/resource/apigateway.py]: Explanation why you can not mix and match properties with and without default parameters:
Python requires defining properties with default values at the end.
If you want to mix properties with and without default values, there is only one way: handle all of them as keyword arguments. This is possible with `attrs` using the `kw_only=True` property on `@define`.
More information here: https://www.attrs.org/en/stable/examples.html?highlight=kw_only#keyword-only-attributes

**Review by aquamatthias — APPROVED** [review]
thanks!

**Pull request merged**